File: Code/src/main/java/com/jmd/web/controller/WebSubmitController.java
package com.jmd.web.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import com.jmd.web.common.RESTfulResult;
import com.jmd.entity.controller.WebDownloadSubmitVo;
import com.jmd.web.service.WebSubmitService;

@CrossOrigin
@RestController
@RequestMapping("/submit")
public class WebSubmitController {
	
	@Autowired
	private WebSubmitService webSubmitService;

	@RequestMapping(value = "blockDownload", method = RequestMethod.POST)
	@ResponseBody
	public RESTfulResult<?> blockDownload(@RequestBody WebDownloadSubmitVo vo) {
		RESTfulResult<?> result = new RESTfulResult<>();
		webSubmitService.blockDownload(vo);
		result.setCode(200);
		result.setMessage("ok");
		result.setSuccess(true);
		return result;
	}

	@RequestMapping(value = "worldDownload", method = RequestMethod.POST)
	@ResponseBody
	public RESTfulResult<?> worldDownload(@RequestBody WebDownloadSubmitVo vo) {
		RESTfulResult<?> result = new RESTfulResult<>();
		webSubmitService.worldDownload(vo);
		result.setCode(200);
		result.setMessage("ok");
		result.setSuccess(true);
		return result;
	}

}

File: Code/src/main/java/com/jmd/web/controller/WebSubmitController.kt
package com.jmd.web.controller

import com.jmd.model.controller.WebDownloadSubmitVo
import com.jmd.web.common.RESTfulResult
import com.jmd.web.service.WebSubmitService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@CrossOrigin
@RestController
@RequestMapping("/submit")
class WebSubmitController {

    @Autowired
    private val webSubmitService: WebSubmitService? = null

    @RequestMapping(value = ["/blockDownload"], method = [RequestMethod.POST])
    @ResponseBody
    fun blockDownload(@RequestBody vo: WebDownloadSubmitVo?): RESTfulResult<*> {
        val result: RESTfulResult<*> = RESTfulResult<Any>()
        webSubmitService!!.blockDownload(vo)
        result.code = 200
        result.message = "ok"
        result.success = true
        return result
    }

    @RequestMapping(value = ["worldDownload"], method = [RequestMethod.POST])
    @ResponseBody
    fun worldDownload(@RequestBody vo: WebDownloadSubmitVo?): RESTfulResult<*> {
        val result: RESTfulResult<*> = RESTfulResult<Any>()
        webSubmitService!!.worldDownload(vo)
        result.code = 200
        result.message = "ok"
        result.success = true
        return result
    }
}